### ISSUE TITLE
add tau (τ)

### ISSUE THREAD

**Issue by alexmo1997** [Qalculate/libqalculate#188]
https://tauday.com/tau-manifesto

**Comment by foreachthing** [comment]
How about adding a custom function?
I think it's easier and faster to just write `2*pi`...

**Comment by alexmo1997** [comment]
Many equations are simpler when you use `τ`, and especially for radians, where e.g. `1/4` of a circle is `1/4τ` or `τ/4`, it's much easier to understand and intuitive than `π` is. It also has a decent amount of agreement and is present in many programming languages (e.g. python, zig, nightly rust, et. al).

**Comment by hanna-kn** [comment]
There are potential conflicts with other uses of tau, which makes me hesitate. Tau currently refers to the tau lepton in libqalculate, but only for the mass of the particle and with an attached subscript.

Tau might be easier to understand and more intuitive than pi, but I doubt that anyone with trouble comprehending and using pi has even heard of tau.

The primary question is, does anybody actually use tau instead of pi?

> How about adding a custom function?

A custom variable, with exact value 2*pi, would be more appropriate.

**Comment by alexmo1997** [comment]
> There are potential conflicts with other uses of tau, which makes me hesitate. Tau currently refers to the tau lepton in libqalculate, but only for the mass of the particle and with an attached subscript.

Fair point, though I think that tau might actually be more well-known that this (might also just be my filter bubble)

>Tau might be easier to understand and more intuitive than pi, but I highly doubt that anyone with trouble comprehending and using pi has even heard of tau.
The primary question is, does anybody actually use tau instead of pi?

I have heared of teachers teaching the trigonometry basics with tau first which got pretty good results, it was over all easier understood than with pi.
Also I do use tau pretty regulary when doing stuff, just not in libqalculate since it doesn't exist there. For me it's not just about understanding it, it's also more trivial to use.

**Comment by hanna-kn** [comment]
It is difficult for me to determine how widespread the actually use of tau, as a pi alternative, is. Google calculator recognize it, but not Wolfram alpha (as a mathematical constant tau is interpreted as the golden ratio).

I will add it to libqalculate (it will likely not do any harm), but reserve the right to remove it if alternative uses are deemed more important. 

I guess there should ideally also be an option to use tau instead of pi in exact results, but I will not make it a priority.

**Comment by jerzydziewierz** [comment]
anyone that has read the tau manifesto (linked in the first post) will recognize that "tau, the circle constant" is a much, much better use for this high-quality symbol than any other "historical" uses. 

quote: "(...) it is the neophytes I am most worried about, for they take the brunt of the damage (made by the adoption of pi)"

**Comment by yashpalgoyal1304** [comment]
Offtopic:

> anyone that has read the tau manifesto that `tau, the circle constant` is a much, much better use for this high-quality symbol

* i haven't read the manifesto, but have seen the Vihart's video. 
* and while i agree, that having the circle constant be `C / r` instead of `C / d` i.e. `6.28...` is much more intuitive, 

+ but i always wish that pi should had that value, with tau being half of it
+ since pi rhymes with full pie, and tau looks exactly like half of pi

this is the reason i hate tau being double of pi, and am rather sticking with status quo, even if it's bad :(

**Comment by jerzydziewierz** [comment]
> * but i always wish that pi should had that value, with tau being half of it

This is not a bad idea as itself, 

Just be aware that we are already fighting against 100+ years of "traditional use of the symbol pi" and hence, just introducing tau into general use would already be a huge win, 

of course, you can always use pi=6.28... in your publications. I honestly wish you best luck. 

In the meantime, it would be great if we had a tau symbol on the keyboard (or somewhere easy to access) in Qalculate, at all.

**Comment by yashpalgoyal1304** [comment]
> I honestly wish you the best of luck.

oh lol 🤣

**Comment by jerzydziewierz** [comment]
> There are potential conflicts with other uses of tau, which makes me hesitate. Tau currently refers to the tau lepton in libqalculate, but only for the mass of the particle and with an attached subscript.
> 
> Tau might be easier to understand and more intuitive than pi, but I highly doubt that anyone with trouble comprehending and using pi has even heard of tau.
> 
> The primary question is, does anybody actually use tau instead of pi?
> 
> > How about adding a custom function?
> 
> A custom variable, with exact value 2*pi, would be more appropriate.



> The primary question is, does anybody actually use tau instead of pi?

I am using it, 

I am advocating for it, 

The best counter-argument for not having it is "it is not in a wide-spread use" -- and the answer is -- of course, it will not be until we begin to use it. Then, it will be used.
